pyomo/core/tests/unit/test_numpy_expr.py

```diff
@@ -0,0 +1,78 @@
+#  ___________________________________________________________________________
+#
+#  Pyomo: Python Optimization Modeling Objects
+#  Copyright 2017 National Technology and Engineering Solutions of Sandia, LLC
+#  Under the terms of Contract DE-NA0003525 with National Technology and
+#  Engineering Solutions of Sandia, LLC, the U.S. Government retains certain
+#  rights in this software.
+#  This software is distributed under the 3-clause BSD License.
+#  ___________________________________________________________________________
+#
+import pyomo.common.unittest as unittest
+
+from pyomo.common.dependencies import (
+    numpy as np, numpy_available,
+    pandas as pd, pandas_available,
+)
+
+from pyomo.environ import (
+    ConcreteModel, Var, RangeSet, Param, Objective, Set, Constraint,
+)
+from pyomo.core.expr.current import MonomialTermExpression
+from pyomo.core.expr.numvalue import NumericNDArray
+
+@unittest.skipUnless(numpy_available, 'numpy is not available')
+class TestNumPy(unittest.TestCase):
+    def test_numpy_scalar_times_scalar_var(self):
+        # Test issue #685
+        m = ConcreteModel()
+        m.x = Var()
+        e = np.float64(5) * m.x
+        self.assertIs(type(e), MonomialTermExpression)
+        self.assertEqual(str(e), "5.0*x")
```

Comment:
I would recommend using the `compare_expressions` function from `pyomo/core/expr/compare.py` throughout these tests.

Reply:
Good call.  Done.